app/views/opportunities/_index_long.html.haml

```diff
@@ -27,16 +27,33 @@
         = t(:added_ago, value: timeago(opportunity.created_at)).html_safe
 
     %dt
-      %b= number_to_currency(opportunity.weighted_amount, precision: 0) + " | "
+      - if opportunity.weighted_amount != 0
+        %b= number_to_currency(opportunity.weighted_amount, precision: 0) + " | "
       - won_or_lost = %w(won lost).include?(opportunity.stage)
       - unless won_or_lost
-        = number_to_currency(opportunity.amount.to_f, precision: 0)
-        - if opportunity.discount
+        - case opportunity.value_to_display
+        - when :all
+          = number_to_currency(opportunity.amount.to_f, precision: 0)
           = t(:discount_number, number_to_currency(opportunity.discount, precision: 0))
-        - else
+          = t(:probability_number, (opportunity.probability || 0).to_s + '%')
+        - when :amount_discount
+          = number_to_currency(opportunity.amount.to_f, precision: 0)
+          = t(:discount_number, number_to_currency(opportunity.discount, precision: 0))
+        - when :amount_probability
+          = number_to_currency(opportunity.amount.to_f, precision: 0)
           = t(:no_discount)
-        = t(:probability_number, opportunity.probability.to_i.to_s + '%') + " | "
-
+          = t(:probability_number, (opportunity.probability || 0).to_s + '%')
+        - when :discount_probability
+          = t(:discount_number, number_to_currency(opportunity.discount, precision: 0))
+          = t(:probability_number, (opportunity.probability || 0).to_s + '%')
+        - when :amount
+          = number_to_currency(opportunity.amount.to_f, precision: 0)
+        - when :discount
+          = t(:discount) + ' ' + number_to_currency(opportunity.discount, precision: 0)
+        - when :probability
+          = t(:probability) + ' ' + opportunity.probability.to_s + '%'
+        - unless opportunity.value_to_display
+          = " | "
```

Comment:
@DmitryAvramec this is not a good implementation, it is highly complex and if we add another field here we'll have to continue to add more permutations.